projects/plugins/jetpack/modules/masterbar/admin-menu/class-wpcom-admin-menu.php

```diff
@@ -250,18 +250,42 @@ public function add_stats_menu() {
 		add_menu_page( __( 'Stats', 'jetpack' ), $menu_title, 'edit_posts', 'https://wordpress.com/stats/day/' . $this->domain, null, 'dashicons-chart-bar', 3 );
 	}
 
+	/**
+	 * Holds the current product, set by get_current_product().
+	 *
+	 * @var array
+	 */
+	private $cached_product = array();
+	/**
+	 * Gets the current product and stores it in $cached_product so the database is only called once per request.
+	 *
+	 * @return array
+	 */
+	private function get_current_product() {
+		// Declare array variable if it does not exist.
+		$current_blog_id = get_current_blog_id();
+		if ( ! isset( $this->cached_product[ $current_blog_id ] ) ) {
+			$this->cached_product[ $current_blog_id ] = null;
```

Comment:
What are the circumstances where there would be more than one site's plan cached in this property?

Reply:
I think this was my suggestion to key against blog id, thinking was to protect against `switch_to_blog` but I admit that is an extreme circumstance to trigger a problem.

Reply:
I feel like we should be ok without the blog id keys, but I have no problem with being overly cautious. 😅 Removing the keys simplifies the function for sure.

Reply:
This code is meant to run within the context of a site so we should be safe to simplify here.